homeassistant/components/homeassistant_yellow/__init__.py

```diff
@@ -0,0 +1,35 @@
+"""The Home Assistant Yellow integration."""
+from __future__ import annotations
+
+from homeassistant.components.hassio import get_os_info
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up a Home Assistant Yellow config entry."""
+    if (os_info := get_os_info(hass)) is None:
+        # The hassio integration has not yet fetched data from the supervisor
+        raise ConfigEntryNotReady
+
+    board: str
+    if (board := os_info.get("board")) is None or not board == "yellow":
+        # Not running on a Home Assistant Yellow, Home Assistant may have been migrated
+        hass.async_create_task(hass.config_entries.async_remove(entry.entry_id))
+        return False
+
+    await hass.config_entries.flow.async_init(
+        "zha",
+        context={"source": "onboarding"},
+        data={
+            "radio_type": "efr32",
+            "port": {
+                "path": "/dev/ttyAMA1",
+                "baudrate": 115200,
+                "flow_control": "hardware",
+            },
+        },
+    )
```

Comment:
Calling this from `async_setup_entry` means we'll try to re-setup zha if the user removes it.
Maybe that's not wanted?

Reply:
The flow now has a confirmation step if onboarding has already been done

Reply:
It would be nice if the user can ignore the config entry if they don't want it.